src/internet_identity/src/http/metrics.rs

```diff
@@ -189,3 +123,92 @@ fn encode_metrics(w: &mut MetricsEncoder<Vec<u8>>) -> std::io::Result<()> {
+            w.encode_gauge(
+                "internet_identity_daily_active_anchors",
+                daily_active_anchor_stats.counter as f64,
+                "The number of unique active anchors in the last completed 24h collection window.",
+            )?;
+            w.encode_gauge(
+                "internet_identity_daily_active_anchors_start_timestamp_seconds",
+                Duration::from_nanos(daily_active_anchor_stats.start_timestamp).as_secs() as f64,
+                "Timestamp of the last completed 24h collection window for unique active anchors.",
+            )?;
+        }
+        if let Some(ref monthly_active_anchor_stats) = stats.completed.monthly_events {
+            w.encode_gauge(
+                "internet_identity_monthly_active_anchors",
+                monthly_active_anchor_stats.counter as f64,
+                "The number of unique active anchors in the last completed 30-day collection window.",
+            )?;
+            w.encode_gauge(
+                "internet_identity_monthly_active_anchors_start_timestamp_seconds",
+                Duration::from_nanos(monthly_active_anchor_stats.start_timestamp).as_secs() as f64,
+                "Timestamp of the last completed 30-day collection window for unique active anchors.",
+            )?;
+        }
+    };
+    if let Some(ref stats) = persistent_state.domain_active_anchor_stats {
+        const BOTH_DOMAINS: &str = "both_ii_domains";
+        labelled_activity_metrics(
+            w,
+            stats,
+            "internet_identity_daily_active_anchors_by_domain",
+            "The number of unique active anchors in the last completed 24h collection window aggregated by II domains used.",
+            "internet_identity_monthly_active_anchors_by_domain",
+            "The number of unique active anchors in the last completed 30-day collection window aggregated by II domains used.",
+            |counter, encoder| {
+                encoder
+                    .value(&[("domain", IC0_APP_DOMAIN)], counter.ic0_app_counter as f64)?
+                    .value(&[("domain", INTERNETCOMPUTER_ORG_DOMAIN)], counter.internetcomputer_org_counter as f64)?
+                    .value(&[("domain", BOTH_DOMAINS)], counter.both_ii_domains_counter as f64)?;
+                Ok(())
+            })?;
+    };
+    if let Some(delegation_origins_limit) = persistent_state.max_num_latest_delegation_origins {
+        w.encode_gauge(
+            "internet_identity_max_num_latest_delegation_origins",
+            delegation_origins_limit as f64,
+            "The maximum number of latest delegation origins that were used with II bound devices.",
+        )?;
+    }
+    Ok(())
+}
+
+fn labelled_activity_metrics<T: ActivityCounter>(
+    w: &mut MetricsEncoder<Vec<u8>>,
+    stats: &ActivityStats<T>,
+    daily_stats_label: &str,
+    daily_stats_doc: &str,
+    monthly_stats_label: &str,
+    monthly_stats_doc: &str,
```

Comment:
maybe package these arguments into a struct, so that one doesn't have to remember the order?